### PR TITLE
Add setuptools to dev-requirements to fix versioneer error

### PR DESCRIPTION
**Pull request by sanders41**
Closes <issue>

### Code Changes

* [x] Add setuptools >= 64.0.2 to dev requirements

### Steps to Confirm

* [ ] run `nox -s "build(dev)"`

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

setuptools 64.0.2 includes a fix to the editable install error with versioneer.

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@ pylint==2.6.0
 pytest==7.1.2
 pytest-cov==3.0.0
 requests-mock==1.9.3
+setuptools>=64.0.2
```
Comment (ThomasLaPiana) [inline dev-requirements.txt:12]: should we more strictly pin this to avoid a similar situation in the future?

**Comment by ThomasLaPiana** [comment]
I'm not sure why but my build works totally fine on main, checked the container and setuptools is at version 58 something 🤷

File: dev-requirements.txt (hunk 1 above)
Comment (sanders41) [reply dev-requirements.txt:12]: I considered this, but we were getting it by default from pip before and I didn't know if we would start causing version conflicts with pip if we do that.

**Review by ThomasLaPiana — APPROVED** [review]
Thanks for the quick fix 👍

File: dev-requirements.txt (hunk 1 above)
Comment (ThomasLaPiana) [reply dev-requirements.txt:12]: good call, hopefully this should be a very rare occurence!

**Pull request merged**